kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -327,14 +327,6 @@ protected TransactionImpl buildTransactionInternal(
     Optional<Protocol> newProtocol = updatedProtocolAndMetadata._1;
     Optional<Metadata> newMetadata = updatedProtocolAndMetadata._2;
 
-    if (!latestSnapshot.isPresent()) {
-      // For now, we generate an empty snapshot (with version -1) for a new table. In the future,
-      // we should define an internal interface to expose just the information the transaction
-      // needs instead of the entire SnapshotImpl class. This should also let us avoid creating
-      // this fake empty initial snapshot.
-      latestSnapshot = Optional.of(getInitialEmptySnapshot(engine, baseMetadata, baseProtocol));
```

Comment:
can we delete the initial snapshot code?

Reply:
oh yeah.. I meant to :)